OMPlot/OMPlotGUI/PlotWindow.cpp

```diff
@@ -39,6 +39,7 @@
 #include "qwt_plot_renderer.h"
 #endif
 #include "qwt_scale_draw.h"
+#include <string>
```

Comment:
Maybe this is the only line that makes the difference???

Reply:
you mean the single line conversion will work just by adding the header?

Reply:
I would guess not. It should simply be that: `const char *s = qtString.toStdString().c_str();` means that s is undefined. Because the std::string might be deallocated after the assignment.

Reply:
@sjoelund but the full expression finishes after `c_str()`, right?

Reply:
Yes, but the `c_str()` results point to internal data of the `std::string` object. It might be removed.

Reply:
Oooucchhh then we need to change at several places :)

Reply:
@adeas31 I don't think too many places are involved, because this is only invalid if you save the `char *` pointer to a variable and in a later step use it (with dangling pointer) If you instead write
```if (!(arrElement = read_csv_dataset(csvReader, varNameQS.toStdString().c_str()))) break;```
everything is fine!

Reply:
I just grepped in OMEdit and OMNotebook and did not find any problematic line, only the 4 lines in OMPlot.